main_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package section is responsible for splitting a program into its different
+// sections without attempting to parse the contents.s
+package main
+
+import (
+	"bytes"
+	"go/token"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewArgParser(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Success -d", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		arg := []string{"-d", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"}
+		_, err := argParser.ParseArgs(arg)
+		require.NoError(t, err)
+		patchFileName := arg[2]
+		assert.Equal(t, true, opts.Diff)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+	})
+	t.Run("Success --diff", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		arg := []string{"--diff", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"}
+		patchFileName := arg[2]
+		_, err := argParser.ParseArgs(arg)
+		require.NoError(t, err)
+		assert.Equal(t, true, opts.Diff)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+	})
+
+}
+
+func TestLoadPatches(t *testing.T) {
+	t.Parallel()
+	t.Run("Success", func(t *testing.T) {
+		opts := &options{Patches: []string{"testdata/patch/error.patch"},
+			Diff:           true,
+			DisplayVersion: false,
+			Verbose:        false,
+		}
+		opts.Args.Patterns = []string{"testdata/test_files/lint_example/test2.go"}
+		patch, err := loadPatches(token.NewFileSet(), opts, bytes.NewReader(nil))
+		assert.Equal(t, 1, len(patch))
```

Comment:
Can use assert.len here.
```suggestion
		assert.Len(t, patch, 1)
```